File: Problems/Int_To_Roman/int_to_roman.py
class solution:
    def int_to_roman(self, num: int) -> str:
        roman: str = ''
        
        while num > 0:
            if num // 1000 > 0:
                fac = num // 1000
                roman += fac * 'M'
                num %= 1000
            elif num // 100 > 0:
                fac = num // 100
                if fac >= 9:
                    roman += 'CM'
                    num -= 900
                elif fac >= 5:
                    roman += 'D'
                    num -= 500
                elif fac >= 4:
                    roman += 'CD'
                    num -= 400
                else:
                    roman += fac * 'C'
                    num -= 100 * fac
            elif num // 10 > 0:
                fac = num // 10
                if fac >= 9:
                    roman += 'XC'
                    num -= 90
                elif fac >= 5:
                    roman += 'L'
                    num -= 50
                elif fac >= 4:
                    roman += 'XL'
                    num -= 40
                else:
                    roman += fac * 'X'
                    num -= fac * 10
            elif num // 1 > 0:
                if num >= 9:
                    roman += 'IX'
                    num -= 9
                elif num >= 5:
                    roman += 'V'
                    num -= 5
                elif num >= 4:
                    roman += 'IV'
                    num -= 4
                else:
                    roman += 'I' * num
                    num -= num
            
                
        return roman
    
    
if __name__ == '__main__':
    num = 1287
    print(solution().int_to_roman(num))

File: Problems/Int_To_Roman/IntToRoman.java
public class IntToRoman {
    public static String intToRoman(int num) {
        StringBuilder roman = new StringBuilder();

        while (num > 0) {
            if (num / 1000 > 0) {
                int fac = num / 1000;
                for (int i = 0; i < fac; i++) roman.append('M');
                num %= 1000;
            } else if (num / 100 > 0) {
                int fac = num / 100;
                if (fac >= 9) {
                    roman.append("CM");
                    num -= 900;
                } else if (fac >= 5) {
                    roman.append('D');
                    num -= 500;
                } else if (fac >= 4) {
                    roman.append("CD");
                    num -= 400;
                } else {
                    for (int i = 0; i < fac; i++) roman.append('C');
                    num -= 100 * fac;
                }
            } else if (num / 10 > 0) {
                int fac = num / 10;
                if (fac >= 9) {
                    roman.append("XC");
                    num -= 90;
                } else if (fac >= 5) {
                    roman.append('L');
                    num -= 50;
                } else if (fac >= 4) {
                    roman.append("XL");
                    num -= 40;
                } else {
                    for (int i = 0; i < fac; i++) roman.append('X');
                    num -= fac * 10;
                }
            } else { // num / 1 > 0
                if (num >= 9) {
                    roman.append("IX");
                    num -= 9;
                } else if (num >= 5) {
                    roman.append('V');
                    num -= 5;
                } else if (num >= 4) {
                    roman.append("IV");
                    num -= 4;
                } else {
                    for (int i = 0; i < num; i++) roman.append('I');
                    num = 0;
                }
            }
        }
        return roman.toString();
    }

    public static void main(String[] args) {
        System.out.println(intToRoman(1287)); // MCCLXXXVII

    }
}
